docs/style.md

```diff
@@ -0,0 +1,188 @@
+
+## Controls
+
+### 4. Do not wrap controls in conditional statements
+
+Reason: This will create dynamic profiles whose controls depend on the execution. The problem here is that we cannot render the profile or provide its information before scanning a system. We want to be able to inform users of the contents of their profiles before they run them. It is valid to skip controls that are not necessary for a system, as long as you do it via `only_if` conditions. Ruby's internal conditionals will hide parts of the profile to static analysis and should thus be avoided.
+
+Avoid:
+```ruby
+if package('..').installed?
+  control "package-test1" do
+    ..
+  end
+end
+```
+
+Use:
+```ruby
+control "package-test1" do
+  only_if { package('..').installed? }
+end
+```
+
+Avoid:
+```ruby
+case inspec.platform.name
+when /centos/
+  include_controls 'centos-profile'
+...
+```
+Use:
+- Set the `supports` attribute in `inspec.yml` files of the profile you want to include:
+```ruby
+supports:
+- platform-name: centos
+```
+Now whenever you run the base profile you can just `include_controls 'centos-profile'`.
+It will only run the included profiles is the platform matches the supported platform.
+
+
+### 5. Do not include dynamic elements in the control IDs
+
+Reason: Control IDs are used to map test results to the tests and profiles. Dynamic control IDs make it impossible to map results back, since the identifier which connects tests and results may change in the process.
+
+Avoid:
+```ruby
+control "test-file-#{name}" do
+  ..
+end
+```
+
+Use:
+```ruby
+control "test-all-files" do
+  ..
+end
+```
+
+Sometimes you may create controls from a static list of elements. If this list stays the same no matter what system is scanned, it may be ok to do so and use it as a generator for static controls.
+
+
+### 6. Avoid Ruby system calls
+
+Reason: Ruby code is executed on the system that runs InSpec. This allows
+InSpec to work without Ruby and rubygems being required on remote
+targets (servers or containers). System calls are often used to interact with
+the local OS or remote endpoints from a local installation.
+InSpec tests, however, are designed to be universally executable on all
+types of runtimes, including local and remote execution. We want to give
+users the ability to take an OS profile and execute it remotely or locally.
+
+**Avoid shelling out**
+
+Avoid: `` `ls``\`
+Avoid: `system("ls")`
+Avoid: `IO.popen("ls")`
+Use: `command("ls")` or `powershell("..")`
+
+Ruby's command executors will only run localy. Imagine a test like this:
+
+```ruby
+describe `whoami` do
+  it { should eq "bob\n" }
+end
+```
+
+If you run this test on your local system and happen to be using Bob's account
+it will succeed. But if you were to run it against `--target alice@remote-host.com`
+it will still report that the user is bob instead of alice.
+
+Instead, do this:
+
+```ruby
+describe command('whoami') do
+  its(:stdout) { should eq "bob\n" }
```

Comment:
We use `its('stdout')` in all other examples.

Reply:
good point!